---
title: "8.5 Ejercicios « Errores y excepciones"
tags: ""
---

### Ejercicio 1

Localiza el error en el siguiente bloque de código. Crea una excepción para evitar que el programa se bloquee y además explica en un mensaje al usuario la causa y/o solución:

```python
resultado = 10/0
```

### Ejercicio 2

Localiza el error en el siguiente bloque de código. Crea una excepción para evitar que el programa se bloquee y además explica en un mensaje al usuario la causa y/o solución:

```python
lista = [1, 2, 3, 4, 5]
lista[10]
```

### Ejercicio 3

Localiza el error en el siguiente bloque de código. Crea una excepción para evitar que el programa se bloquee y además explica en un mensaje al usuario la causa y/o solución:

```python
colores = { 'rojo':'red', 'verde':'green', 'negro':'black' } 
colores['blanco']
```

### Ejercicio 4

Localiza el error en el siguiente bloque de código. Crea una excepción para evitar que el programa se bloquee y además explica en un mensaje al usuario la causa y/o solución:

```python
resultado = 15 + "20"
```

### Ejercicio 5

Realiza una función llamada agregar_una_vez(lista, el) que reciba una lista y un elemento. La función debe añadir el elemento al final de la lista con la condición de no repetir ningún elemento. Además si este elemento ya se encuentra en la lista se debe invocar un error de tipo ValueError que debes capturar y mostrar este mensaje en su lugar:

`Error: Imposible añadir elementos duplicados => [elemento].`

Cuando tengas la función intenta añadir los siguiente valores a la lista 10, -2, "Hola" y luego muestra su contenido.

> Sugerencia:
> Puedes utilizar la sintaxis "elemento in lista"

```python
elementos = [1, 5, -2]

# Completa el ejercicio aquí
```
